source/includes/fact-call-limitations.rst

```diff
@@ -0,0 +1,104 @@
+.. Top level title in calling page, won't render in right TOC
+
+The results of database queries cannot be passed inside the following
+contexts:
+
+- Class constructor functions
+- Non-async generator functions
+- Callbacks to ``.sort()`` on an array
+
+To access to the results of database calls, use `async functions
+<https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function>`__,
+`async generator functions
+<https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for-await...of>`__,
+or ``.map()``.
+
+Constructors
+~~~~~~~~~~~~
+
+The following constructors do not work: 
+
+.. code-block:: javascript
+   :copyable: false
+   
+   // This code will fail
+   class FindResults {
+      constructor() {
+         this.value = db.students.find();
+      }
+   }
+
+   // This code will fail
+   function listEntries() { return db.students.find(); }
+   class FindResults {
+      constructor() {
+         this.value = listEntries();
+      }
+   }
+
+Use an ``async`` function instead:
+
+.. code-block:: javascript
+
+   class FindResults {
+      constructor() {
+         this.value = ( async() => {
+            return db.students.find();
+         } )();
+      }
+   }
```

Comment:
It *might* also be worth adding a second alternative here, e.g.:

```js
class FindResults {
  constructor() {
  }

  init() {
    this.value = db.students.find();
  }
}
```

This would mean that consumers of this class would have to call `.init()` after already having constructed the class instance, but the big advantage would be that there is no knowledge of asynchronous code in JS required in order for this to work